packages/conventional-changelog-writer/test/index.spec.js

```diff
@@ -851,6 +852,70 @@ describe('conventionalChangelogWriter', function () {
     })
   })
 
+  describe('generate loose semver', function () {
+    function getStream () {
+      var upstream = through.obj()
+      upstream.write({
+        header: 'feat(scope): broadcast $destroy event on scope destruction',
+        body: null,
+        footer: null,
+        notes: [],
+        references: [],
+        committerDate: '2015-04-07 14:17:05 +1000'
+      })
+      upstream.write({
+        header: 'fix(ng-list): Allow custom separator',
+        body: 'bla bla bla',
+        footer: null,
+        notes: [],
+        references: [],
+        version: '1.0.01-1',
+        committerDate: '2015-04-07 15:00:44 +1000'
+      })
+      upstream.write({
+        header: 'perf(template): tweak',
+        body: 'My body.',
+        footer: null,
+        notes: [],
+        references: [],
+        committerDate: '2015-04-07 15:01:30 +1000'
+      })
+      upstream.end()
+
+      return upstream
+    }
+
+    it('should generate on `\'version\'` if it\'s at least a valid loose semver and loose option is passed', function (done) {
+      var i = 0
+
+      getStream()
+        .pipe(conventionalChangelogWriter({ version: '1.1.01-1' }, { looseSemver: true }))
```

Comment:
I haven't read through this code in a while, so I may be missing this: Is there a test that validates expected behavior when `version: '1.1.01-1'` is set and `looseSemver` is `false`?

Reply:
Let me dig into it again and figure out. I think passing the version directly to the writer will still allow the full version to be written (with the leading zero in this case), it'll just fail validation for seeing if it's a patch version.